datacats/cli/pull.py

```diff
@@ -38,7 +38,20 @@ def pull(opts):
                      you are going offline.
 """
     for i in IMAGES + (EXTRA_IMAGES if opts['--all'] else []):
-        pull_image(i)
+        retrying_pull_image(i)
+
+
+def retrying_pull_image(image_name):
+    for retries in range(1, 6):
+        try:
+            pull_image(image_name)
+            break
+        except DatacatsError:
+            if retries <= 5:
+                print
+                print 'Error while pulling image {}. Retrying.'.format(image_name)
+            else:
+                raise
```

Comment:
still wrong, I think. This will fail silently because retries never reaches the value 6. Maybe try writing this as a function that you can add a test for to see if it actually works